test/spec/OktaUserAgent.ts

```diff
@@ -1,63 +1,118 @@
+const SDK_VERSION = (global as any).SDK_VERSION;
+const NODE_VERSION = (global as any).NODE_VERSION;
+
 import { OktaUserAgent } from '../../lib/http/OktaUserAgent';
 
-jest.mock('../../lib/features', () => {
+const mocked = {
+  isBrowser: jest.fn()
+};
+jest.mock('lib/features', () => {
   return {
-    isBrowser: () => {}
+    __esModule: true,
+    isBrowser: () => mocked.isBrowser()
   };
 });
 
-const mocked = {
-  features: require('../../lib/features')
-};
-
 describe('OktaUserAgent', () => {
-  let oktaUserAgent;
-  let sdkVersion;
-  beforeEach(async () => {
-    sdkVersion = (await import('../../package.json')).version;
-    oktaUserAgent = new OktaUserAgent();
+  const context: any = {};
+
+  beforeEach(() => {
+    context.oktaUserAgent = new OktaUserAgent();
   });
 
   describe('browser env', () => {
     beforeEach(() => {
-      jest.spyOn(mocked.features, 'isBrowser').mockReturnValue(true);
+      mocked.isBrowser.mockReturnValue(true);
+      context.expected = `okta-auth-js/${SDK_VERSION}`;
+      context.oktaUserAgent = new OktaUserAgent();
     });
+
     it('gets okta-auth-js and node info in the Okta UA by default', () => {
+      const { oktaUserAgent, expected } = context;
       const httpHeader = oktaUserAgent.getHttpHeader();
       expect(httpHeader).toEqual({ 
-        'X-Okta-User-Agent-Extended': `okta-auth-js/${sdkVersion}`
+        'X-Okta-User-Agent-Extended': expected
       });
     });
+
     it('can add extra environment', () => {
+      const { oktaUserAgent, expected } = context;
       oktaUserAgent.addEnvironment('fake/x.y');
       const httpHeader = oktaUserAgent.getHttpHeader();
       expect(httpHeader).toEqual({ 
-        'X-Okta-User-Agent-Extended': `okta-auth-js/${sdkVersion} fake/x.y` 
+        'X-Okta-User-Agent-Extended': `${expected} fake/x.y`
+      });
+    });
+
+    // Reason: OKTA-641280
+    it('should return same header after multiple calls', () => {
+      const { oktaUserAgent, expected } = context;
+      expect(oktaUserAgent.getHttpHeader()).toEqual({
+        'X-Okta-User-Agent-Extended': expected
+      });
+      expect(oktaUserAgent.getHttpHeader()).toEqual({
+        'X-Okta-User-Agent-Extended': expected
+      });
+      expect(oktaUserAgent.getHttpHeader()).toEqual({
+        'X-Okta-User-Agent-Extended': expected
+      });
+      expect(oktaUserAgent.getHttpHeader()).toEqual({
+        'X-Okta-User-Agent-Extended': expected
+      });
+      expect(oktaUserAgent.getHttpHeader()).toEqual({
+        'X-Okta-User-Agent-Extended': expected
       });
     });
   });
 
   describe('node env', () => {
-    let nodeVersion = process.versions.node;
     beforeEach(() => {
-      jest.spyOn(mocked.features, 'isBrowser').mockReturnValue(false);
+      mocked.isBrowser.mockReturnValue(false);
+      (global as any).node = NODE_VERSION;
```

Comment:
What does this line do? As I understand, Node version is being read from `process.versions.node`

Reply:
That line actually does nothing, I just tested it 🤦

Reply:
remove before merging if it's not doing anything